File: FrontEnd/src/app/features/reservations/reservations/classes/dtos/form/reservation-read-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { CustomerAutoCompleteVM } from 'src/app/features/reservations/customers/classes/view-models/customer-autocomplete-vm'
import { Metadata } from 'src/app/shared/classes/metadata'
import { PassengerReadDto } from './passenger-read-dto'
import { PickupPointAutoCompleteVM } from '../../../../pickupPoints/classes/view-models/pickupPoint-autocomplete-vm'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface ReservationReadDto extends Metadata {

    reservationId: Guid
    customer: CustomerAutoCompleteVM
    destination: SimpleEntity
    driver: SimpleEntity
    pickupPoint: PickupPointAutoCompleteVM
    port: SimpleEntity
    portAlternate: SimpleEntity
    ship: SimpleEntity
    date: string
    refNo: string
    email: string
    phones: string
    remarks: string
    adults: number
    kids: number
    free: number
    totalPax: number
    ticketNo: string
    passengers: PassengerReadDto

}

File: API/Features/CheckIn/Dtos/ReservationReadDto.cs
using System.Collections.Generic;
using API.Infrastructure.Classes;

namespace API.Features.CheckIn {

    public class ReservationReadDto {

        // PK
        public string ReservationId { get; set; }
        // Fields
        public string Date { get; set; }
        public string RefNo { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public int TotalPax { get; set; }
        public string TicketNo { get; set; }
        public string Email { get; set; }
        public string Phones { get; set; }
        public string Remarks { get; set; }
        // Metadata
        public string PutAt { get; set; }
        //  Navigation
        public SimpleEntity Customer { get; set; }
        public SimpleEntity Destination { get; set; }
        public PickupPointReadDto PickupPoint { get; set; }
        public List<PassengerReadDto> Passengers { get; set; }

    }

}